### PR TITLE
[agent/DB] Prevent sql injection by using `sqlite3_bind_text`

### PR DESCRIPTION
**Pull request by anyj0527**
- Use `sqlite3_bind_text` instead of formatted string to prevent sql injection attack.

### CODE REVIEW

File: daemon/service-db.cc [hunk 1]
```diff
@@ -279,27 +274,18 @@ MLServiceDB::get_pipeline (const std::string name, std::string &description)
...
 
-error:
   if (value) {
     description = std::string (value);
     g_free (value);
   } else {
-    throw std::invalid_argument ("Failed to get pipeline description.");
+    throw std::invalid_argument ("Failed to get pipeline description of " + name);
   }
 }
```
Comment (songgot) [inline daemon/service-db.cc:290]: I think it needs "sqlite3_finalize (res)".

File: daemon/service-db.cc [hunk 2]
```diff
@@ -279,27 +274,18 @@ MLServiceDB::get_pipeline (const std::string name, std::string &description)
...
 
   sqlite3_finalize (res);
 
-error:
   if (value) {
     description = std::string (value);
     g_free (value);
   } else {
-    throw std::invalid_argument ("Failed to get pipeline description.");
+    throw std::invalid_argument ("Failed to get pipeline description of " + name);
```
Comment (songgot) [inline daemon/service-db.cc:288]: How about as below?
```
MLServiceDB::get_pipeline (const std::string name, std::string &description)
{
  char *value = nullptr;
  sqlite3_stmt *res;

  if (name.empty ())
    throw std::invalid_argument ("Invalid name parameters!");
  std::string key_with_prefix = DB_KEY_PREFIX;
  key_with_prefix += name;

  if (sqlite3_prepare_v2 (_db, "SELECT description FROM tblPipeline WHERE key = ?1", -1, &res, nullptr) != SQLITE_OK ||
      sqlite3_bind_text (res, 1, key_with_prefix.c_str (), -1, NULL) != SQLITE_OK ||
      sqlite3_step (res) != SQLITE_ROW) {
    sqlite3_finalize (res);
    throw std::invalid_argument ("Failed to get pipeline description of " + name);
  }

  value = g_strdup_printf ("%s", sqlite3_column_text (res, 0));
  sqlite3_finalize (res);

  description = std::string (value);
  g_free (value);
}
```

File: daemon/service-db.cc [hunk 3]
```diff
@@ -279,27 +274,18 @@ MLServiceDB::get_pipeline (const std::string name, std::string &description)
   std::string key_with_prefix = DB_KEY_PREFIX;
   key_with_prefix += name;
```
Comment (songgot) [inline daemon/service-db.cc:276]: How about add 
```
  if (name.empty () || description.empty ())
    throw std::invalid_argument ("Invalid name or value parameters!");
```

File: daemon/service-db.cc [hunk 4]
```diff
@@ -509,24 +467,13 @@ MLServiceDB::activate_model (const std::string name, const guint version)
...
-
-  rc = sqlite3_step (res);
-  if (rc == SQLITE_ROW) {
-    int count = sqlite3_column_int (res, 0);
-    if (count != 1) {
-      g_critical ("There is no model with name %s and version %u", name.c_str (), version);
-      sqlite3_finalize (res);
-      throw std::invalid_argument ("There is no model with the given name and version.");
-    }
+  if (sqlite3_prepare_v2 (_db, "SELECT EXISTS(SELECT 1 FROM tblModel WHERE key = ?1 AND version = ?2)", -1, &res, nullptr) != SQLITE_OK ||
```
Comment (songgot) [inline daemon/service-db.cc:470]: How about add ?
```
  if (name.empty ())
    throw std::invalid_argument ("Invalid name");
```

File: daemon/service-db.cc [hunk 5]
```diff
@@ -588,36 +534,27 @@ MLServiceDB::get_model (const std::string name, std::string &model, const gint v
   key_with_prefix += name;
```
Comment (songgot) [inline daemon/service-db.cc:535]: How about check name and model ?

File: daemon/service-db.cc (hunk 3 above)
Comment (anyj0527) [reply daemon/service-db.cc:276]: `description` is reference of std::string. I think empty check of it is unnecessary.

File: daemon/service-db.cc (hunk 5 above)
Comment (songgot) [reply daemon/service-db.cc:535]: model is a  reference, Thank you.

**Review by songgot — APPROVED** [review]
Exception handling is well done and looks like good code.
LGTM

**Review by jaeyun-jung — APPROVED** [review]

**Pull request merged**